integration/spark2/src/main/scala/org/apache/spark/sql/execution/datasources/SparkCarbonTableFormat.scala

```diff
@@ -353,7 +353,8 @@ private class CarbonOutputWriter(path: String,
   def writeCarbon(row: InternalRow): Unit = {
     val data = new Array[AnyRef](fieldTypes.length + partitionData.length)
     var i = 0
-    while (i < fieldTypes.length) {
+    val fieldTypesLen = fieldTypes.length
+    while (i < fieldTypesLen) {
```

Comment:
Is it really has difference after the change for lots of columns?

Reply:
@jackylk  I have test it and add description in https://issues.apache.org/jira/browse/CARBONDATA-3626

Reply:
> Is it really has difference after the change for lots of columns?

It can significantly improve load data into carbon table with lots of columns.